File: back-end/01-ms-identity/src/main/java/com/faceattend_edu/identity_service/adapter/in/web/dto/AuthRequest.java
package com.faceattend_edu.identity_service.adapter.in.web.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class AuthRequest {

    @NotBlank(message = "username is required")
    private String username;

    @NotBlank(message = "password is required")
    private String password;
}

File: front-end/Mobile/src/models/identity/AuthRequest.js
import BaseModel from '../BaseModel';

export default class AuthRequest extends BaseModel {
  constructor(data = {}) {
    super();
    this.username = data.username || data.email || '';
    this.password = data.password || '';
  }

  static fromCredentials(username, password) {
    return new AuthRequest({ username, password });
  }

  toApi() {
    return {
      username: this.username,
      password: this.password,
    };
  }
}
